meson.build

```diff
@@ -0,0 +1,1859 @@
+    features += audiounit.name()
+    sources += files('audio/out/ao_audiounit.m')
+    audiounit += {'use': true}
+elif get_option('audiounit').enabled()
+    error('audiounit enabled but it could not be found!')
+endif
+
+coreaudio = dependency('appleframeworks', modules: ['CoreFoundation', 'CoreAudio', 
+                       'AudioUnit', 'AudioToolbox'], required: get_option('coreaudio'))
+if coreaudio.found()
+    dependencies += coreaudio
+    features += coreaudio.name()
+    sources += files('audio/out/ao_coreaudio.c',
+                     'audio/out/ao_coreaudio_exclusive.c',
+                     'audio/out/ao_coreaudio_properties.c')
+endif
+
+if audiounit['use'] or coreaudio.found()
+    sources += files('audio/out/ao_coreaudio_chmap.c',
+                     'audio/out/ao_coreaudio_utils.c')
+endif
+
+jack = dependency('jack', required: get_option('jack'))
+if jack.found() and get_option('gpl')
+    dependencies += jack
+    features += jack.name()
+    sources += files('audio/out/ao_jack.c')
+endif
+
+openal = dependency('openal', version: '>= 1.13', required: get_option('openal'))
+if openal.found()
+    dependencies += openal
+    features += openal.name()
+    sources += files('audio/out/ao_openal.c')
+endif
+
+opensles = cc.find_library('OpenSLES', required: get_option('opensles'))
+if opensles.found()
+    dependencies += opensles
+    features += 'opensles'
+    sources += files('audio/out/ao_opensles.c')
+endif
+
+oss = false
+if get_option('oss-audio').allowed()
+    oss = cc.has_header_symbol('sys/soundcard.h', 'SNDCTL_DSP_SETPLAYVOL')
+endif
+if oss
+    features += 'oss-audio'
+    sources += files('audio/out/ao_oss.c')
+elif get_option('oss-audio').enabled()
+    error('oss-audio enabled but it could not be found!')
+endif
+
+pulse = dependency('libpulse', version: '>= 1.0', required: get_option('pulse'))
+if pulse.found()
+    dependencies += pulse
+    features += pulse.name()
+    sources += files('audio/out/ao_pulse.c')
+endif
+
+sdl2_audio = sdl2.found() and get_option('sdl2-audio').allowed()
+if sdl2_audio
+    features += 'sdl2-audio'
+    sources += files('audio/out/ao_sdl.c')
+elif get_option('sdl2-audio').enabled()
+    error('sdl2-audio enabled but sdl2 was not found!')
+endif
+
+wasapi = cc.has_header_symbol('audioclient.h', 'IAudioClient') and win32 and \
+         get_option('wasapi').allowed()
+if wasapi
+    features += 'wasapi'
+    sources += files('audio/out/ao_wasapi.c',
+                     'audio/out/ao_wasapi_changenotify.c',
+                     'audio/out/ao_wasapi_utils.c')
+elif get_option('wasapi').enabled()
+    error('wasapi enabled but it could not be found!')
+endif
+
+
+# video output dependencies
+caca = dependency('caca', version: '>= 0.99.beta18', required: get_option('caca'))
+if caca.found()
+    dependencies += caca
+    features += caca.name()
+    sources += files('video/out/vo_caca.c')
+endif
+
+direct3d = {
+    'name': 'direct3d',
+    'header': cc.check_header('d3d9.h'),
```

Comment:
check_header does support required too. I guess direct3d['use'] should be equal to `direct3d['header'] and win32_desktop and get_option('gpl')`